src/cls/rbd/cls_rbd_types.h

```diff
@@ -73,13 +74,81 @@ struct MirrorImage {
   static void generate_test_instances(std::list<MirrorImage*> &o);
 
   bool operator==(const MirrorImage &rhs) const;
+  bool operator<(const MirrorImage &rhs) const;
 };
 
 std::ostream& operator<<(std::ostream& os, const MirrorImageState& mirror_state);
 std::ostream& operator<<(std::ostream& os, const MirrorImage& mirror_image);
 
 WRITE_CLASS_ENCODER(MirrorImage);
 
+enum MirrorImageStatusState {
+  MIRROR_IMAGE_STATUS_STATE_UNKNOWN         = 0,
+  MIRROR_IMAGE_STATUS_STATE_ERROR           = 1,
+  MIRROR_IMAGE_STATUS_STATE_SYNCING         = 2,
+  MIRROR_IMAGE_STATUS_STATE_STARTING_REPLAY = 3,
+  MIRROR_IMAGE_STATUS_STATE_REPLAYING       = 4,
+  MIRROR_IMAGE_STATUS_STATE_STOPPING_REPLAY = 5,
+  MIRROR_IMAGE_STATUS_STATE_STOPPED         = 6,
+};
+
+enum MirrorImageStatusFlag {
+  MIRROR_IMAGE_STATUS_FLAG_UP               = 1<<0,
+  MIRROR_IMAGE_STATUS_FLAG_DOWN             = 1<<1,
```

Comment:
Probably better to have this and the API match.  The only future-proofing that I can think of right now is for when we add horizontal scaling to rbd-mirroring.  If we have an orderly transition of one image from rbd-mirror instance A to instance B, it might be nice to avoid showing a warning.  However, I don't think adding a flag will help us in that situation.